en/02_Developer_Guides/14_Files/05_File_Manipulation.md

```diff
@@ -0,0 +1,419 @@
+            throw new LogicException('Cannot convert ' . $this->getOwner()->appCategory() . ' to image.');
+        }
+
+        $original = $this->getOwner();
+        return $original->manipulateExtension(
+            $newExtension,
+            function (AssetStore $store, string $filename, string $hash, string $variant) {
+                $tmpFilePath = /* some conversion logic goes here */;
+                $backend = Injector::inst()->create(Image_Backend::class);
+                $backend->loadFrom($tmpFilePath);
+                $tuple = $backend->writeToStore(
+                    $store,
+                    $filename,
+                    $hash,
+                    $variant,
+                    ['conflict' => AssetStore::CONFLICT_USE_EXISTING]
+                );
+                return [$tuple, $backend];
+            }
+        );
+    }
+
+    /**
+     * Create a variant of the file as a pdf.
+     *
+     * @throws LogicException if the conversion is not supported
+     */
+    public function toPdf(): DBFile
+    {
+        if (!$this->getIsConversionSupported('pdf')) {
+            throw new LogicException('Cannot convert ' . $this->getOwner()->appCategory() . ' to pdf.');
+        }
+
+        $original = $this->getOwner();
+        return $file->manipulateExtension(
+            'pdf',
+            function (AssetStore $store, string $filename, string $hash, string $variant) {
+                $tmpFilePath = /* some conversion logic goes here */;
+                $tuple = $store->setFromLocalFile(
+                    $tmpFilePath,
+                    $filename,
+                    $hash,
+                    $variant,
+                    ['conflict' => AssetStore::CONFLICT_USE_EXISTING]
+                );
+                return [$tuple, null];
+            }
+        );
+    }
+
+    private function getIsConversionSupported(string $convertTo): bool
+    {
+        return in_array($this->getOwner()->appCategory(), self::SUPPORT_CONVERSION_FROM_CATEGORY[$convertTo]);
+    }
+}
+```
+
+After applying the extension to both the `File` and `DBFile` classes, you can use these methods in PHP or in templates.
+
+```yml
+SilverStripe\Assets\File:
+  extensions:
+    - App\Extension\ImageFormatExtension
+SilverStripe\Assets\Storage\DBFile:
+  extensions:
+    - App\Extension\ImageFormatExtension
+```
+
+Okay, now lets step through those and take a look at what's going on. We'll only look at the parts that are different from the image-to-image conversion [we looked at earlier](#converting-between-image-formats).
+
+#### Converting something to an image
+
+The main difference between converting between images, and converting a non-image to an image, is that you have to get a third-party to perform the conversion for you.
+
+```php
+$tmpFilePath = /* some conversion logic goes here */;
+$backend = Injector::inst()->create(Image_Backend::class);
+$backend->loadFrom($tmpFilePath);
```

Comment:
This is following the same way I'm walking developers through the code in the "Converting something to an image" and "Converting between image formats" sections. It's a nice way to break down the big chunk of code, without inundating the code with some very verbose commentary.

Reply:
> This is following the same way I'm walking developers through the code in the "Converting something to an image"

This IS the "Converting something to an image" section :-)

> Converting between image formats

I've recommended we don't include that whole section because it's redundant as soon as we implement the conversion code

Reply:
> This IS the "Converting something to an image" section :-)

Sorry, thought this was one down. My point still stands.

> I've recommended we don't include that whole section because it's redundant as soon as we implement the image conversion code in core

I disagree with that recommendation - and even if I followed it, I'd just have to convert basically all of that doc to here instead because it's needed to explain the core concepts at play.